ts/src/apex.ts

```diff
@@ -0,0 +1,1940 @@
+            'id': this.safeString (income, 'id'),
+            'amount': this.safeNumber (income, 'fundingValue'),
+            'rate': this.safeNumber (income, 'rate'),
+        };
+    }
+
+    /**
+     * @method
+     * @name apex#setLeverage
+     * @description set the level of leverage for a market
+     * @see https://api-docs.pro.apex.exchange/#privateapi-v3-for-omni-post-sets-the-initial-margin-rate-of-a-contract
+     * @param {float} leverage the rate of leverage
+     * @param {string} symbol unified market symbol
+     * @param {object} [params] extra parameters specific to the exchange API endpoint
+     * @returns {object} response from the exchange
+     */
+    async setLeverage (leverage: Int, symbol: Str = undefined, params = {}) {
+        if (symbol === undefined) {
+            throw new ArgumentsRequired (this.id + ' setLeverage() requires a symbol argument');
+        }
+        await this.loadMarkets ();
+        const market = this.market (symbol);
+        const leverageString = this.numberToString (leverage);
+        const initialMarginRate = Precise.stringDiv ('1', leverageString, 4);
+        const request: Dict = {
+            'symbol': market['id'],
+            'initialMarginRate': initialMarginRate,
+        };
+        const response = await this.privatePostV3SetInitialMarginRate (this.extend (request, params));
+        const data = this.safeDict (response, 'data', {});
+        return data;
+    }
+
+    /**
+     * @method
+     * @name apex#fetchPositions
+     * @description fetch all open positions
+     * @see https://api-docs.pro.apex.exchange/#privateapi-v3-for-omni-get-retrieve-user-account-data
+     * @param {string[]} [symbols] list of unified market symbols
+     * @param {object} [params] extra parameters specific to the exchange API endpoint
+     * @returns {object[]} a list of [position structure]{@link https://docs.ccxt.com/#/?id=position-structure}
+     */
+    async fetchPositions (symbols: Strings = undefined, params = {}) {
+        const response = await this.privateGetV3Account (params);
+        const data = this.safeDict (response, 'data', {});
+        const positions = this.safeList (data, 'positions', []);
+        return this.parsePositions (positions, symbols);
+    }
+
+    parsePosition (position: Dict, market: Market = undefined) {
+        //
+        // {
+        //     "symbol": "BTC-USDT",
+        //     "status": "",
+        //     "side": "LONG",
+        //     "size": "0.000",
+        //     "entryPrice": "0.00",
+        //     "exitPrice": "",
+        //     "createdAt": 1690366452416,
+        //     "updatedTime": 1690366452416,
+        //     "fee": "0.000000",
+        //     "fundingFee": "0.000000",
+        //     "lightNumbers": "",
+        //     "customInitialMarginRate": "0"
+        // }
+        const marketId = this.safeString (position, 'symbol');
+        market = this.safeMarket (marketId, market);
+        const symbol = market['id'];
```

Comment:
upon parsing the response, we should always use `market['symbol']`  so that we return the unified symbol in the unified response